src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java

```diff
@@ -97,4 +100,32 @@ public boolean deleteProgrammeMembershipsForTrainee(String traineeTisId) {
 
     return true;
   }
+
+  /**
+   * Sign Condition of Joining with the given programme membership ID.
+   *
+   * @param programmeMembershipId The ID of the programme membership for signing COJ.
+   * @return The updated programme membership
+   *     or empty if the programme membership with the ID was not found.
+   */
+  public Optional<ProgrammeMembership> signProgrammeMembershipCoj(String programmeMembershipId) {
+
+    TraineeProfile traineeProfile = repository.findByProgrammeMembershipId(programmeMembershipId);
```

Comment:
We should use the existing `findByTraineeTisId()` repository method, then filter based on the programme membership ID. As it stands there is nothing stopping a user from signing someone else's CoJ.

Reply:
I see, It needs to have TraineeTisId pass in to the API as well if it is what you mean?
Something like this: `/api/trainee/programme-membership/{traineeTisId}/{programmeMembershipId}/sign-coj`?

Reply:
That still relies on trusting the user to not give us someone else's `traineeTisId`.

Have a look at `GET: /api/trainee-profile`, it gets the `custom:tisId` value from the auth token. We should do the same for this new endpoint.

Reply:
Nice one. I got it. Thank you @Judge40 !